### DISCUSSION TITLE
Tamagui / Solito not using ".native.ts" file for cross platform support

### DISCUSSION THREAD

**Discussion by kennethburlabountytms** [tamagui/tamagui#2669] [General]
Hey guys just wanna ask help regarding the issue so i have two file to cover both web (next js) and native (react native). However tamagui is using .ts file instead of .native.ts file. To be honest I'm still doing my research since i don't know what's really causing this issue, but just in case any one is familiar with this issue. Thanks in advance 🙏

Packages:
- react-native: 0.71.3
- tamagui: 1.8.4
- expo: 48.0.7
- solito: 3.0.0
- next: 13.1.6

<img width="286" alt="Screenshot 2024-05-22 at 10 13 00 PM" src="https://github.com/tamagui/tamagui/assets/147617726/51d6d642-5ead-4a2b-b60e-0dc0a199094b">
<img width="314" alt="Screenshot 2024-05-22 at 10 13 13 PM" src="https://github.com/tamagui/tamagui/assets/147617726/f5a001fc-afee-421a-9212-51f11b8d82de">

**Comment by wallynm** [comment]
@kennethburlajoinbountyai, i had a similar issue with this last day related with Supabase import that only should been imported into .native files. 

In order to use `.native` first you need to [ensure those files aren't inside `App` folder](https://docs.expo.dev/router/advanced/platform-specific-modules/), Expo doesn't handle native extensions inside `App folder`. If this doesn't solve the problem, you can try to create a `index.ts` file, to handle the importing part - in my case importing directly the component.native.ts (component.ts) wasn't working, to solve it i needed to first create this index file, which was responsible for handling the native code, in the end this actually solved my problem (don't know why).

Seems that expo wasn't recognizing correctly before setting as this.
